server/py/services/api/crud/model_monitoring/grafana.py

```diff
@@ -291,122 +112,90 @@ async def grafana_overall_feature_analysis(
-        services.api.crud.ModelEndpoints().get_model_endpoint,
-        project=project,
-        endpoint_id=endpoint_id,
-    )
-
-    time_series = []
+    # Filters
+    model = query_parameters.get("model", None)
+    function = query_parameters.get("function", None)
 
-    feature_names = endpoint.spec.feature_names
+    uids = (
+        query_parameters.get("uids", "").split(",")
+        if query_parameters.get("uids")
+        else None
+    )
 
-    if not feature_names:
-        logger.warn(
-            "'feature_names' is either missing or not initialized in endpoint record",
-            endpoint_id=endpoint.metadata.uid,
-        )
-        return time_series
+    labels = query_parameters.get("labels", "")
+    labels = labels.split(",") if labels else []
 
-    path = config.model_endpoint_monitoring.store_prefixes.default.format(
-        project=project, kind=mlrun.common.schemas.ModelMonitoringStoreKinds.EVENTS
-    )
-    _, container, path = parse_model_endpoint_store_prefix(path)
+    # Endpoint type filter - will be used to filter the router models
+    filter_router = query_parameters.get("filter_router", None)
 
-    client = get_frames_client(
-        token=auth_info.data_session,
-        address=config.v3io_framesd,
-        container=container,
+    endpoint_list = await services.api.crud.ModelEndpoints().list_model_endpoints(
+        db_session=db_session,
+        project=project,
+        model_name=model,
+        function_name=function,
+        labels=labels,
+        uids=uids,
+        tsdb_metrics=True,
     )
 
-    data: pd.DataFrame = await run_in_threadpool(
-        client.read,
-        backend="tsdb",
-        table=path,
-        columns=feature_names,
-        filter=f"endpoint_id=='{endpoint_id}'",
-        start=start,
-        end=end,
+    allowed_endpoints = await framework.utils.auth.verifier.AuthVerifier().filter_project_resources_by_permissions(
+        mlrun.common.schemas.AuthorizationResourceTypes.model_endpoint,
+        endpoint_list.endpoints,
+        lambda _endpoint: (
+            _endpoint.metadata.project,
+            _endpoint.metadata.uid,
+        ),
+        auth_info,
     )
+    endpoint_list.endpoints = allowed_endpoints
 
-    data.drop(["endpoint_id"], axis=1, inplace=True, errors="ignore")
-    data.index = data.index.astype(np.int64) // 10**6
+    table = grafana_schemas.GrafanaModelEndpointsTable()
+    for endpoint in endpoint_list.endpoints:
+        if (
+            filter_router
+            and endpoint.status.endpoint_type
+            == mlrun.common.schemas.model_monitoring.EndpointType.ROUTER
+        ):
+            continue
+        row = [
+            endpoint.metadata.uid,
+            endpoint.metadata.name,
+            endpoint.spec.function_name,
+            endpoint.spec.model_name,
+            endpoint.spec.model_class,
+            endpoint.status.error_count,
+            endpoint.status.result_status,
```

Comment:
why to use this? and not to retrieve  it from frames?

Reply:
In grafana dashboard we can't use multiple datasources on the same chart. Now, in this case these values are being retrieved as part of a summary table of the model endpoints which is through `MLRun API` datasource and not `frames`. Unfortunately that also means that these values are not affect by the selected time range (we explain in the chart docs).